models/Device.js

```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+const mongooseStringQuery = require('mongoose-string-query');
+
+/**
+This file creates a model and fills it with a new instance.
+*/
+const DeviceSchema = new mongoose.Schema({
+  id: {
+    type: String,
+  },
+  name: {
+    type: String,
+  },
+});
+
+DeviceSchema.plugin(mongooseStringQuery);
+
+/**
+ * Creates a new device from the data sent in the request
+ * @param {Object} data - Data sent in request
+ * @returns {Object} - Object with all the information for a new Device
+ */
+
+function createDevice(data) {
+  try {
+    if (!data.id) throw new Error('No ID provided.');
+    if (!data.name) throw new Error('Device name not provided.');
```

Comment:
I remember why this puzzled me: I am of the opinion that your routes have a specification and should take on the job of sanitizing inputs before passing them as arguments to your app logic, i.e. the problem of verifying id and name should fall on the route, not the model.